astroid/nodes/node_classes.py

```diff
@@ -2523,11 +2523,16 @@ class ExceptHandler(mixins.MultiLineBlockMixin, mixins.AssignTypeMixin, Statemen
     _astroid_fields = ("type", "name", "body")
     _multi_line_block_fields = ("body",)
 
+    # TODO: Make parent non-optional
+    parent: Optional["nodes.TryExcept"]
```

Comment:
Saying the `parent` attribute is required (except for Module nodes), might work with a decorator.
```py
    def __init__(..., parent: Optional["NodeNG"], ...):
        self._parent = parent  # this needs to stay optional

    @property
    def parent(self) -> "NodeNG":
        assert self._parent  # here, we make sure it has actually been assigned
        return self._parent

    @parent.setter
    def parent(self, value: "NodeNG") -> None:
        self._parent = value
```

If you would like to explore that further in a new PR, an additional tip. Since `Module` inherits from `NodeNG` and doesn't have a parent, the decorator can't be added there. You'd probably need to create a new subclass which all nodes (except for `Module`) inherit from to implement it.